### PR TITLE
Feature/mark

### PR DESCRIPTION
**Pull request by marklnichols**
This is the first pass at displaying values for intermediate function calls. The function added to Main.hs should result in this UI:

**Source**    

    g :: forall a . Integral a => a -> a -> Double
    g x y =    
       let least = lcm x y
       in fromIntegral least ^^ gcd x y

**Variables**
- $arg1 = 6
- $arg2 = 15
- $result = 27000.0
- fromIntegral = 30.0
- _gcd = \<function>
- _gcd' = 3
- _lcm = \<function>
- _lcm' = 30
- least = 30
- x = 6
- y = 15

I disabled infix function handling for this PR, I realized I have to do some substitution to create legal variable names for some of the operator functions.

The couple of tests I added should be automated hspec, etc. tests, but I left them as just 'prints' for now to avoid breaking your build.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
 /dist/
 /stack.yaml
+ghcid.txt
```
Comment (ndmitchell) [inline .gitignore]: How come you are ignoring `ghcid.txt`? What are you putting in there?

File: src/Debug.hs [hunk 2]
```diff
@@ -50,15 +50,18 @@ module Debug(
     debugView, debugSave, debugPrint,
     -- * Clear a trace
     debugClear,
+    -- * Exported for tests only
+    removeLet,
+    removeExtraDigits
```
Comment (ndmitchell) [inline src/Debug.hs]: The `Debug` module name is very important to keep small, so I suggest breaking these into `Debug.Util` or similar and importing them into `Debug` and the test.

File: .gitignore (hunk 1 above)
Comment (marklnichols) [reply .gitignore]: Only because i didn't see it I the original repo and figured you didn't want it

File: src/Debug.hs (hunk 2 above)
Comment (marklnichols) [reply src/Debug.hs]: Ok, I'll do that tonight

**Comment by ndmitchell** [comment]
I should say more generally that it looks quite plausible, thanks a lot. I've got the flu right now so not in a detail state of mind but will get to it soon.

File: .gitignore (hunk 1 above)
Comment (ndmitchell) [reply .gitignore]: I don't. I just wondered why you had it? Are you using VS Code and using that for the IDE integration? If so, just use "Start GHCid" in VS Code and it does all that for you.

Comment (marklnichols) [reply .gitignore]: Oh, yes I was using the original method. I'll give the 'start' a try

**Comment by marklnichols** [comment]
Ok, feel better.

**Comment by ndmitchell** [comment]
Thanks - looks good!

File: .gitignore (hunk 1 above)
Comment (marklnichols) [reply .gitignore]: I found the Start GHCid command and it worked.  But the instructions might need an update as they still mention creating the text file.

**Pull request merged**